### PR TITLE
add Olive exception skeleton

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
Sometimes, the quantization would fail https://dev.azure.com/aiinfra/Model%20optimization%20Toolkit/_build/results?buildId=321080&view=logs&j=83e9455c-f768-5dd6-ffb8-4d16293cb8a9&t=8c02a2da-1821-5532-3cad-c631fe1ed2f8 

Olive should have the ability to handle this kind of error not caused by Olive bug. when this happens, the quantization should be ignored. 

As the result, we introduce OliveException system to allow Olive throw the ignorable exception in pass run

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/engine/engine.py [hunk 1]
```diff
@@ -817,6 +822,9 @@ def _run_passes(
             else:
                 signal = self._evaluate_model(model, model_id, evaluator_config, accelerator_spec)
             logger.debug(f"Signal: {signal}")
+        else:
+            signal = None
+            logger.warning("Skipping evaluation as model was pruned")
```
Comment (trajepl) [inline olive/engine/engine.py:827]: As the `signal` will be set anytime, maybe we can remove the Line 851.

File: olive/engine/engine.py [hunk 2]
```diff
@@ -883,6 +891,9 @@ def _run_pass(
                 input_model = self._prepare_non_local_model(input_model)
             try:
                 output_model = host.run_pass(p, input_model, output_model_path, pass_search_point)
+            except OliveException as e:
```
Comment (trajepl) [inline olive/engine/engine.py:894]: Better to OlivePassException? As we aim to catch the exception from pass run.

File: olive/exception/__init__.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class OlivePassException(OliveException):
+    """
+    Base class for Olive pass exceptions.
+    """
+
+    pass
+
+
+class OliveEvaluationException(OliveException):
```
Comment (trajepl) [inline olive/exception/__init__.py:21]: Will we need the evaluation exception class? As there is no any `prune` mechanism in evaluation part.

Comment (guotuofeng) [reply olive/exception/__init__.py:21]: this is used to raise if there are some failure when evaluation model.

**Review by jambayk — APPROVED** [review]

**Pull request merged**